src/app/src/hooks.js

```diff
@@ -110,3 +112,46 @@ export function useFilePicker(onChange) {
 
     return openFileDialog;
 }
+
+export function useBoundaryId() {
+    return useParams().boundaryId;
+}
+
+export function useTrailingDebounceCallback({
+    callback,
+    immediateCallback,
```

Comment:
Consider adding a comment here describing the expected difference between `callback` and `immediateCallback`